core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -696,54 +668,47 @@ private TrustAnchor getTrustAnchor(final JsonObject trustedCa) throws GeneralSec
      * 
      * @return This tenant so the API can be used fluently.
      */
-    private TenantObject addTrustedCA(final JsonObject trustedCa) {
+    public TenantObject addTrustedCA(final TrustedCertificateAuthority trustedCa) {
 
         if (!isValidTenantConfig(trustedCa)) {
             throw new IllegalArgumentException("invalid tenant configuration");
         }
 
-        if (trustConfigurations == null) {
-            trustConfigurations = new HashMap<>();
+        if (trustedAuthorities == null) {
+            trustedAuthorities = new ArrayList<>();
         }
 
-        final String subjectDn = getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, String.class);
+        trustedAuthorities.add(trustedCa);
 
-        final List<JsonObject> trustedCas = trustConfigurations.getOrDefault(subjectDn, new ArrayList<>());
-
-        trustedCas.add(trustedCa);
-        trustConfigurations.put(subjectDn, trustedCas);
         return this;
     }
 
-    private Instant getNotBefore(final JsonObject trustedCa) {
-        Instant instant = null;
-        final String strValue = getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_NOT_BEFORE, String.class);
-        if (strValue != null) {
-            instant = DateTimeFormatter.ISO_INSTANT.parse(strValue, Instant::from);
-        }
-        return instant;
-    }
-
-    private Instant getNotAfter(final JsonObject trustedCa) {
-        Instant instant = null;
-        final String strValue = getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_NOT_AFTER, String.class);
-        if (strValue != null) {
-            instant = DateTimeFormatter.ISO_INSTANT.parse(strValue, Instant::from);
-        }
-        return instant;
-    }
-
-    private boolean isValidTenantConfig(final JsonObject trustedCa) {
+    private boolean isValidTenantConfig(final TrustedCertificateAuthority trustedCa) {
 
-        final String subjectDn = getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, String.class);
-        final String cert = getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_CERT, String.class);
-        final String publicKey = getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_PUBLIC_KEY, String.class);
+        final byte[] cert = trustedCa.getCertificate();
+        final byte[] publicKey = trustedCa.getPublicKey();
 
-        if (Strings.isNullOrEmpty(subjectDn) ||
-                // public key or the encoded certificate must be present
-                Strings.isNullOrEmpty(cert) && Strings.isNullOrEmpty(publicKey)) {
+        // public key or the encoded certificate must be present
+        if (Strings.isNullOrEmpty(cert) && Strings.isNullOrEmpty(publicKey)) {
             return false;
         }
         return true;
     }
+
+    /**
+     * TODO.
+     * @return TODO.
+     */
+    @JsonIgnore
+    public List<JsonObject> getTrustedCAs() {
```

Comment:
what is this method good for? who needs it?

Reply:
its used in the `FileBasedTenantService#convertTenantObject` when converting a `TenantObject` to a `Tenant`. I think the its better to call it `getTrustedCaConfigurations()`?

Reply:
IMHO it should call *getTrustedAuthorities*, shouldn't it? Since we are sharing the `TrustedCertificateAuthority` class there is nothing to convert, is there?

Reply:
> IMHO it should call getTrustedAuthorities, shouldn't it?

there is already such a method which returns a `List<TrustedCertificateAuthority>`. The `getTrustedCAs` method returns the trust configurations in JSON format.

> Since we are sharing the TrustedCertificateAuthority class there is nothing to convert, is there?

But they are of different types. its the entire tenant object thats converted, not just the `TrustedCertificateAuthority`:

```java
    static Tenant convertTenantObject(final TenantObject tenantObject) {

        if (tenantObject == null) {
            return null;
        }

        final var tenant = new Tenant();

        tenant.setEnabled(tenantObject.isEnabled());
       ...
        Optional.ofNullable(tenantObject.getTrustedCAs())
                .ifPresent(trustConfigs -> setTrustedAuthorities(tenant, trustConfigs)); 
      ...
        return tenant;
    }
```

I cannot do `tenant:: setTrustedAuthorities` even if the method returns `org.eclipse.hono.TrustedCertificateAuthority`, because the method expects a different type

Reply:
???
just do
```
tenant.setTrustedAuthorities(new LinkedList(tenantObject.getTrustedAuthorities()));
```